pkg/resource/replication_group/custom_update_api.go

```diff
@@ -691,3 +721,40 @@ func (rm *resourceManager) newModifyReplicationGroupRequestPayload(
 
 	return input
 }
+
+// cacheNodeTypeRequiresUpdate retrieves the last requested cacheNodeType saved in annotations and compares them
+// to the current desired cacheNodeType
+func cacheNodeTypeRequiresUpdate(desired *resource) bool {
+	annotations := desired.ko.ObjectMeta.GetAnnotations()
+	if val, ok := annotations[AnnotationLastRequestedCNT]; ok && desired.ko.Spec.CacheNodeType != nil {
+		return val != *desired.ko.Spec.CacheNodeType
+	}
+
+	// This means there is delta and no value in annotation or in Spec
+	return true
+}
+
+// nodeGroupRequiresUpdate retrieves the last applied NumNodeGroups and NodeGroupConfiguration and compares them
+// to the current desired NumNodeGroups and NodeGroupConfiguration
+func nodeGroupRequiresUpdate(desired *resource) bool {
+	annotations := desired.ko.ObjectMeta.GetAnnotations()
+
+	if val, ok := annotations[AnnotationLastRequestedNNG]; ok && val != "null" {
+		numNodes, _ := strconv.ParseInt(val, 10, 64)
```

Comment:
nit: I personally think that we should not ignore errors in int/json parsing and return an error when it's the case.